usecases/event-driven-pipeline/README.md

```diff
@@ -0,0 +1,304 @@
+- IAM execution roles for EMR on EKS, Step Functions and EventBridge 
+
+![terraform-output](img/terraform-output-eb-sfn-ack.png)
+
+### Validate
+
+The following command will update the `kubeconfig` on your local machine and allow you to interact with your EKS Cluster using `kubectl` to validate the deployment.
+
+### Run `update-kubeconfig` command:
+
+```bash
+aws eks --region us-west-2 update-kubeconfig --name event-driven-pipeline-demo
+```
+
+### List the nodes
+
+```bash
+kubectl get nodes
+
+# Output should look like below
+NAME                                        STATUS   ROLES    AGE     VERSION
+ip-10-1-10-64.us-west-2.compute.internal    Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-10-65.us-west-2.compute.internal    Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-10-7.us-west-2.compute.internal     Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-10-73.us-west-2.compute.internal    Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-11-96.us-west-2.compute.internal    Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-12-197.us-west-2.compute.internal   Ready    <none>   19h     v1.24.9-eks-49d8fe8
+```
+
+## Set up the event-driven pipeline
+### Create an EMR virtual cluster
+Let’s start by creating a [virtual cluster](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/virtual-cluster.html) in EMR and link it with a Kubernetes namespace in EKS. By doing that, the virtual cluster will use the linked namespace in EKS for hosting spark workloads.
+
+Let's apply the manifest by using kubectl command below. Once done, you can go to *EMR console*, click *Virtual clusters* and you should see the cluster record as shown below.  
+```bash
+kubectl apply -f ack-yamls/emr-virtualcluster.yaml
+```
+![virtual cluster](img/vc-sfn.png)
+
+
+### Create a S3 bucket and upload data
+Next, let’s create a S3 bucket for storing spark pod templates and sample data. 
+```bash
+kubectl apply -f ack-yamls/s3.yaml
+```
+*Note*: If you don’t see the bucket got created, you can check the log from ACK S3 controller pod for details. The error is mostly caused by the bucket with the same name has existed. You need to change the bucket name in s3.yaml as well as in eventbridge.yaml and sfn.yaml. You also need to update upload-inputdata.sh and upload-spark-scripts.sh with the new bucket name.
+
+Run the command below to upload input data and pod templates. Once done, sparkjob-demo-bucket S3 bucket is created with two folders: input and scripts.
+```bash
+bash spark-scripts-data/upload-inputdata.sh
+```
+
+### Create a Step Functions state machine
+
+You need to make the following changes in sfn.yaml before apply. 
+
+* replace the value for roleARN with stepfunctions_role_arn 
+* replace the value for ExecutionRoleArn with emr_on_eks_role_arn
+* replace the value for VirtualClusterId with your virtual cluster id
+* optional: change sparkjob-demo-bucket with your bucket name 
```

Comment:
I don't think this is optional? If the user doesn't have access to this bucket, then the process will fail?